### PR TITLE
Suggestion: send SIGSTOP to login and it's children

### PR DESCRIPTION
**Pull request by christoph-zededa**
to disable keyboard access

Problem: modules might get loaded automatically and we cannot in all cases unload the module for the USB controller

Idea: send SIGSTOP to login and it's children

### CODE REVIEW

**Comment by rene** [comment]
Comparing to the module loading, I think this solution is better, because it should work also for serial consoles (where we don't rely on usbhid modules). @christoph-zededa , did you think in any situation where this SIGSTP can be awkward? Like, what about if we are running some FS and/or other blocking operations and then we receive this SIGSTP... I'm more afraid about FS operations, that could lead to corruption if the device is powered off...

I think the best approach to control this console login access would be through PAM plugins... we could configure a PAM plugin to communicate with pillar and then we would be able to implement a very fined login control, blocking any login or even defining access by users, etc....

**Comment by christoph-zededa** [comment]
> did you think in any situation where this SIGSTP can be awkward? 
I think it is awkward when one starts a long running process (e.g. a bpftrace script) and then a config update comes in.

But I would hope the user knows what is happening if they do such a change.

> PAM plugins
I only know they're hard to develop and to configure ;-)

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 1]
```diff
@@ -3429,6 +3433,52 @@ func addUSBtoKernel() {
...
+		log.Warnf("could not send signal to login: %v", err)
+
+	}
+
+	childrenProcs, err := loginProc.Children()
+	if err != nil {
+		log.Warnf("could not retrieve children processes: %v", err)
+	}
+	for _, proc := range childrenProcs {
+		err = proc.SendSignal(sig)
```
Comment (christoph-zededa) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:3478]: note for myself: check `err`

File: pkg/pillar/cmd/domainmgr/domainmgr.go [hunk 2]
```diff
@@ -3429,6 +3433,52 @@ func addUSBtoKernel() {
...
+		return
+	}
+
+	err = loginProc.SendSignal(sig)
+	if err != nil {
+		log.Warnf("could not send signal to login: %v", err)
+
+	}
+
+	childrenProcs, err := loginProc.Children()
```
Comment (christoph-zededa) [inline pkg/pillar/cmd/domainmgr/domainmgr.go:3473]: not to myself: does this work even after `eve enter debug`?

**Review by eriknordmark — COMMENTED** [review]
How does this approach related to the debug.enable.console (which enables and disabled getty)?

**Comment by christoph-zededa** [comment]
> How does this approach related to the debug.enable.console (which enables and disabled getty)?

I think that's the better way to do it then; I didn't know this exists.

**Pull request closed without merging**